src/gen_tw.erl

```diff
@@ -14,27 +14,44 @@
          uuid/0
         ]).
 
--export_type([ref/0, event/0]).
+-export_type([ref/0, event/0, virtual_time/0]).
 
 -export([init/4]).
 
 -callback init(Arg::term()) -> {ok, InitialState::term()} | {error, Reason::term()}.
--callback tick_tock(CurrentLVT::integer(), State::term()) -> {NextLVT::integer(), NextState::term()}.
--callback handle_event(CurrentLVT::integer(), EventLVT::integer(), Event::term(), ModuleState::term()) ->
+
+-callback tick_tock(CurrentLVT::virtual_time(), State::term()) -> {NextLVT::virtual_time(), NextState::term()}.
+-callback handle_event(CurrentLVT::virtual_time(), EventLVT::virtual_time(), Event::term(), ModuleState::term()) ->
     {ok, NextState::term()} |
     {error, Reason::term()}.
 -callback terminate(State::term()) -> any().
 
 -record(event,
-    {lvt,           %% Simulation time the event is to be applied
-     id,            %% Unique identifier for the event
-     anti = 0,      %% 0 for event, -1 for antievent
-     link,          %% Causal link for the event (Pid)
-     payload        %% Event payload
+    {lvt     :: virtual_time(),    %% Simulation time the event is to be applied
+     id      :: uuid:uuid(),       %% Unique identifier for the event
+     anti    :: boolean(),         %% false for event, true for antievent
```

Comment:
This won't work.

the value of anti for an antievent must be < the value of the anti field for an actual event.